scheduler/src/cook/rest/api.clj

```diff
@@ -552,20 +542,34 @@
                           ports)]
       {:docker/port-mapping port-maps})))
 
+(defn- ensure-user-parameter
+  "Ensures the presence of the user parameter by attaching it when missing in params."
+  [user id params]
+  (let [user-id (util/user->user-id user)
+        group-id (util/user->group-id user)
```

Comment:
If these return `nil`, we should throw an error.

Reply:
added exception handling